ccl/changefeedccl: TestChangefeedDiff failed

SHA: https://github.com/cockroachdb/cockroach/commits/e1c1a9ae765730782f3adc2298a76d6a07ad055d

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestChangefeedDiff PKG=github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1626698&tab=buildLog

```

I191205 07:24:01.563275 6231 base/addr_validation.go:296  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I191205 07:24:01.563417 6231 base/addr_validation.go:342  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
W191205 07:24:01.569774 6231 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I191205 07:24:01.616527 6231 server/server.go:973  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I191205 07:24:01.623421 6231 server/config.go:573  [n?] 1 storage engine initialized
I191205 07:24:01.623501 6231 server/config.go:576  [n?] RocksDB cache size: 128 MiB
I191205 07:24:01.623537 6231 server/config.go:576  [n?] store 0: in-memory, size 0 B
I191205 07:24:01.624383 6231 server/server.go:1433  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I191205 07:24:01.624466 6231 server/server.go:1443  [n?] **** add additional nodes by specifying --join=127.0.0.1:39103
I191205 07:24:01.650236 6231 server/node.go:333  [n?] **** cluster 8f8bddb8-7607-4b79-b9dd-b2076663f8ad has been created
I191205 07:24:01.650793 6231 gossip/gossip.go:394  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:39103" > attrs:<> locality:<> ServerVersion:<major_val:19 minor_val:2 patch:0 unstable:4 > build_tag:"v20.1.0-alpha.20191118-516-ge1c1a9a" started_at:1575530641650590756 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:39687" > 
W191205 07:24:01.670891 6466 storage/store.go:1518  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I191205 07:24:01.675908 6231 server/node.go:437  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=11 KiB), ranges=24, leases=2, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=166.00 pMax=10260.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I191205 07:24:01.676159 6231 storage/stores.go:240  [n1] read 0 node addresses from persistent storage
I191205 07:24:01.676422 6231 server/node.go:654  [n1] connecting to gossip network to verify cluster ID...
I191205 07:24:01.676550 6231 server/node.go:674  [n1] node connected via gossip and verified as part of cluster "8f8bddb8-7607-4b79-b9dd-b2076663f8ad"
I191205 07:24:01.677012 6231 server/node.go:521  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I191205 07:24:01.677591 6231 server/server.go:2103  [n1] Could not start goroutine dumper worker due to: directory to store dumps could not be determined
I191205 07:24:01.679045 6231 server/server.go:1563  [n1] starting https server at 127.0.0.1:40101 (use: 127.0.0.1:40101)
I191205 07:24:01.679183 6231 server/server.go:1568  [n1] starting postgres server at 127.0.0.1:39687 (use: 127.0.0.1:39687)
I191205 07:24:01.679248 6231 server/server.go:1570  [n1] starting grpc server at 127.0.0.1:39103
I191205 07:24:01.679309 6231 server/server.go:1571  [n1] advertising CockroachDB node at 127.0.0.1:39103
I191205 07:24:01.796716 6568 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I191205 07:24:01.854894 6240 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.2-4 User:root}
I191205 07:24:01.935826 6815 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:d8723166-baf5-4eee-98e1-45d5d6754a99 User:root}
I191205 07:24:01.974604 6899 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
```

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x45738df]

goroutine 6231 [running]:
testing.tRunner.func1(0xc0009aa000)
	/usr/local/go/src/testing/testing.go:830 +0x69d
panic(0x4f8a6a0, 0x8dcf780)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc000a801e0, 0x635d560, 0xc000104010)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:182 +0x155
panic(0x4f8a6a0, 0x8dcf780)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc000a801e0, 0x635d560, 0xc000104010)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:479 +0x705
panic(0x4f8a6a0, 0x8dcf780)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest.(*cloudFeed).walkDir(0xc000c73d50, 0xc0006d62d0, 0x90, 0x0, 0x0, 0x62e7cc0, 0xc0026e7d40, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest/testfeed.go:622 +0x9f
path/filepath.walk(0xc000232540, 0x3a, 0x639c2e0, 0xc000c47ad0, 0xc002a112b8, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:378 +0x264
path/filepath.walk(0xc00197f650, 0x2f, 0x639c2e0, 0xc000c47a00, 0xc002a112b8, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:382 +0x39e
path/filepath.Walk(0xc00197f650, 0x2f, 0xc002a112b8, 0x33, 0xc001c0c988)
	/usr/local/go/src/path/filepath/path.go:404 +0x144
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest.(*cloudFeed).Next(0xc000c73d50, 0xc0000105d0, 0x8e2b420, 0xc0026dc380)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest/testfeed.go:610 +0x984
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.readNextMessages(0x63e6fc0, 0xc0009aa000, 0x637be60, 0xc000c73d50, 0x1, 0xc002a11500, 0x85f30f, 0xc0009aa000, 0xc0013bb1a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:62 +0xf8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase(0x63e6fc0, 0xc0009aa000, 0x637be60, 0xc000c73d50, 0xc0023708d0, 0x1, 0x1, 0xc002370800)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:91 +0xab
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads(0x63e6fc0, 0xc0009aa000, 0x637be60, 0xc000c73d50, 0xc0023708d0, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:102 +0x95
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedDiff.func1(0xc0009aa000, 0xc00016e300, 0x63152a0, 0xc0007a3900)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:135 +0x694
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cloudStorageTest.func1(0xc0009aa000)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:331 +0x747
testing.tRunner(0xc0009aa000, 0xc001898050)
	/usr/local/go/src/testing/testing.go:865 +0x164
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x65b
```

--- a/pkg/ccl/changefeedccl/cdctest/testfeed.go
+++ b/pkg/ccl/changefeedccl/cdctest/testfeed.go
@@ -613,10 +613,20 @@ func (c *cloudFeed) Next() (*TestFeedMessage, error) {
 	}
 }
 
-func (c *cloudFeed) walkDir(path string, info os.FileInfo, _ error) error {
+func (c *cloudFeed) walkDir(path string, info os.FileInfo, err error) error {
 	if strings.HasSuffix(path, `.tmp`) {
 		// File in the process of being written by ExternalStorage. Ignore.
 		return nil
+	}
+
+	if err != nil {
+		// From filepath.WalkFunc:
+		//  If there was a problem walking to the file or directory named by
+		//  path, the incoming error will describe the problem and the function
+		//  can decide how to handle that error (and Walk will not descend into
+		//  that directory). In the case of an error, the info argument will be
+		//  nil. If an error is returned, processing stops.
+		return err
 	}
 
 	if info.IsDir() {

